air/src/proof/mod.rs

```diff
@@ -207,61 +207,126 @@ fn get_conjectured_security(
+                options,
+                base_field_bits,
+                lde_domain_size,
+                trace_domain_size,
+                a as usize,
+            )
+        })
+        .expect(
+            "Should not fail since m_max is larger than m_min
+                             for all trace sizes of length greater than 4",
+        );
 
-    let blowup_bits = options.blowup_factor().ilog2() as f64;
-    let num_fri_queries = options.num_queries() as f64;
-    let lde_size_bits = lde_domain_size.trailing_zeros() as f64;
-
-    // blowup_plus_bits is the number of bits in the blowup factor which is the inverse of
-    // `\rho^+ := (trace_domain_size + 2) / lde_domain_size`. `\rho^+` is used in order to define a larger
-    // agreement parameter `\alpha^+ := (1 + 1/2m)\sqrt{rho^+} := 1 - \theta^+`. The reason for
-    // running FRI with a larger agreement parameter is to account for the simplified
-    // DEEP composition polynomial. See Protocol 3 in https://eprint.iacr.org/2022/1216.
-    let blowup_plus_bits = ((lde_domain_size as f64) / (trace_domain_size as f64 + 2_f64)).log2();
-
-    // m is a parameter greater or equal to 3.
-    // A larger m gives a worse field security bound but a better query security bound.
-    // An optimal value of m is then a value that would balance field and query security
-    // but there is no simple closed form solution.
-    // This sets m so that field security is equal to the best query security for any value
-    // of m, unless the calculated value is less than 3 in which case it gets rounded up to 3.
-    let mut m = extension_field_bits + 1.0;
-    m -= options.grinding_factor() as f64;
-    m -= 1.5 * blowup_bits;
-    m -= 0.5 * num_fri_queries * blowup_plus_bits;
-    m -= 2.0 * lde_size_bits;
-    m /= 7.0;
-    m = 2.0_f64.powf(m);
-    m -= 0.5;
-    m = m.max(3.0);
-
-    // compute pre-FRI query security
-    // this considers only the third component given in the corresponding part of eq. 20
-    // in https://eprint.iacr.org/2021/582, i.e. (m+1/2)^7.n^2 / (2\rho^1.5.q) as all
-    // other terms are negligible in comparison.
-    let pre_query_security = (extension_field_bits + 1.0
-        - 3.0 / 2.0 * blowup_bits
-        - 2.0 * lde_size_bits
-        - 7.0 * (m + 0.5).log2()) as u32;
+    cmp::min(
+        proven_security_protocol_m(
+            options,
+            base_field_bits,
+            lde_domain_size,
+            trace_domain_size,
+            m_optimal as usize,
+        ),
+        collision_resistance as u64,
+    ) as u32
+}
 
-    // compute security we get by executing multiple query rounds
-    let security_per_query = 0.5 * blowup_plus_bits - (1.0 + 1.0 / (2.0 * m)).log2();
-    let mut query_security = (security_per_query * num_fri_queries) as u32;
+/// Computes proven security level for the specified proof parameters for a fixed
+/// value of the proximity parameter m.
+#[cfg(feature = "std")]
+fn proven_security_protocol_m(
```

Comment:
Should we mention that this is in the list-decoding regime?

Also small nit: maybe we should rename this to `get_proven_security_for_m()`?

Reply:
Makes sense!